syntax = "proto3";

package user_svc;

service UserService {
  rpc Auth (AuthRequest) returns (AuthResponse);
  rpc NewUser(NewUserRequest) returns (NewUserResponse);
}

message User {
  string first_name = 1;
  string last_name = 2;
  string username = 3;
  string password = 4;
  string uid = 5;
  int32 age = 6;
  Gender gender = 7;
  Location location = 8;
}

enum Gender {
  Male = 0;
  Female = 1;
}

message Location {
  double longitude = 1;
  double latitude = 2;
}

message AuthRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse {
  string jwt = 1;
}

message NewUserRequest {
  User user = 1;
}

message NewUserResponse {}